list_objects/list_test.go

```diff
@@ -0,0 +1,116 @@
+package list_objects
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+)
+
+func TestList(t *testing.T) {
+	tc := it.StartNewClusterWithOptions(t.Name(), it.NextPort(), 1)
+	defer tc.Shutdown()
+	ctx := context.Background()
+	dc := tc.DefaultConfig()
+	c, err := hazelcast.StartNewClientWithConfig(ctx, dc)
+	require.NoError(t, err)
+	// create two to check ordering of the same type
+	mapName1, mapName2 := fmt.Sprintf("%s-map1", t.Name()), fmt.Sprintf("%s-map2", t.Name())
+	_, err = c.GetMap(ctx, mapName1)
+	require.NoError(t, err)
+	_, err = c.GetMap(ctx, mapName2)
+	require.NoError(t, err)
+	// PN Counter
+	pnCounterName := it.NewUniqueObjectName("pnCounter")
+	_, err = c.GetPNCounter(ctx, pnCounterName)
+	require.NoError(t, err)
+	// List
+	listName := it.NewUniqueObjectName("list")
+	_, err = c.GetList(ctx, listName)
+	require.NoError(t, err)
+	// MultiMap
+	mmapName := it.NewUniqueObjectName("multimap")
+	_, err = c.GetMultiMap(ctx, mmapName)
+	require.NoError(t, err)
+	// Queue
+	queueName := it.NewUniqueObjectName("queue")
+	_, err = c.GetQueue(ctx, queueName)
+	require.NoError(t, err)
+	// FlakeIDGenerator
+	figName := it.NewUniqueObjectName("flakeIDGen")
+	_, err = c.GetFlakeIDGenerator(ctx, figName)
+	require.NoError(t, err)
+	// Replicated Map
+	repMapName := it.NewUniqueObjectName("replicatedMap")
+	_, err = c.GetReplicatedMap(ctx, repMapName)
+	require.NoError(t, err)
+	// Set
+	setName := it.NewUniqueObjectName("set")
+	_, err = c.GetSet(ctx, setName)
+	require.NoError(t, err)
+	// Topic
+	topicName := it.NewUniqueObjectName("topic")
+	_, err = c.GetReplicatedMap(ctx, topicName)
+	require.NoError(t, err)
+	tcs := []struct {
+		name   string
+		args   []string
+		expect []string
+	}{
+		{
+			name: "list all",
+			args: []string{},
+			expect: []string{
+				// todo find out why flake id generators output as "PNCounter"
```

Comment:
I guess these are due to Go Client, not CLC.

Reply:
Also cc @yuce

Reply:
Thanks @yuce, keeping this open until the Go client PR is merged.

Reply:
Go client fix update is here https://github.com/hazelcast/hazelcast-commandline-client/pull/133/commits/c34809fe0c997fb10c232bba01701bbf734a7ae0